### ISSUE TITLE
Multiple Pointclouds with only one sources.json file

### ISSUE THREAD

**Issue by pewixg3** [potree/potree#620]
Hi potree community,

I just saw that a source.json file could have multiple sources but when I integrate the pointcloud with this file, there is only one pointcloud. But I want to have a pointcloud per source. This may be useful to enable or disable pointclouds sources into jstree without create several sources.json files.

Is someone have a solution ?

**Comment by ThomasFarmer** [comment]
why dont you just multiply them in the html file, like this: 
```
		Potree.loadPointCloud("pointclouds/Paks/Tile_00062/cloud.js", "Tile_0062", e => {
			let pointcloud = e.pointcloud;
			let material = pointcloud.material;
			viewer.scene.addPointCloud(pointcloud);
			material.pointColorType = Potree.PointColorType.RGB; // any Potree.PointColorType.XXXX 
			material.size = 1;
			material.pointSizeType = Potree.PointSizeType.ADAPTIVE;
			material.shape = Potree.PointShape.SQUARE;
			viewer.fitToScreen();
		});
		
		Potree.loadPointCloud("pointclouds/Paks/Tile_0063/cloud.js", "Tile_0063", e => {
			let pointcloud = e.pointcloud;
			let material = pointcloud.material;
			viewer.scene.addPointCloud(pointcloud);
			material.pointColorType = Potree.PointColorType.RGB; // any Potree.PointColorType.XXXX 
			material.size = 1;
			material.pointSizeType = Potree.PointSizeType.ADAPTIVE;
			material.shape = Potree.PointShape.SQUARE;
			viewer.fitToScreen();
		});
		Potree.loadPointCloud("pointclouds/Paks/Tile_0118/cloud.js", "Tile_0118", e => {
			let pointcloud = e.pointcloud;
			let material = pointcloud.material;
			viewer.scene.addPointCloud(pointcloud);
			material.pointColorType = Potree.PointColorType.RGB; // any Potree.PointColorType.XXXX 
			material.size = 1;
			material.pointSizeType = Potree.PointSizeType.ADAPTIVE;
			material.shape = Potree.PointShape.SQUARE;
			viewer.fitToScreen();
		});
```
etc... 
in the end its the same, you have 3 (or more) clouds you can switch on and off at will. you can even assign them different default values, like for example you can set the last one to appear not as RNG but with elevation coloring or something. complite your clouds as separate ones without the generate-page tag and refer them individually.

**Comment by pewixg3** [comment]
Thanks for your answer

I know that I can do this but the idea was about to don't spend time to convert several pointclouds but only one
If this isn't possible I will do like you advise me :)

**Comment by ThomasFarmer** [comment]
well, i ain't exactly no expert, but from what i can gather it may be difficult and it may need some modifications in the core code if you wanna do that. you see when you refer a cloud, you call the function `Potree.loadPointCloud `which takes 3 parameters, which you can find in potree.js around line 403 (may vary with version):

```
Potree.loadPointCloud = function (path, name, callback) {
	let loaded = function (pointcloud) {
		pointcloud.name = name;
		callback({type: 'pointcloud_loaded', pointcloud: pointcloud});
	};
```




The second paramter is name, which as you can see in my example is what appears in the node tree ( = the stuff you can check and uncheck). 
![balm](https://user-images.githubusercontent.com/23472718/54211153-1b6d7780-44e1-11e9-8413-aef29010bda5.jpg)
![balm2](https://user-images.githubusercontent.com/23472718/54211159-1dcfd180-44e1-11e9-85d4-8752dbb318bc.jpg)
I am just trying to understand the code's logic here, but to me it seems if you reference one .js file / .json next to it, the least of the issues you would get is that all your clouds would have the same name / reference in the node tree, which is something you clearly want to avoid (based on your core problem description). what could be an alternative is to modify the 
```
    "boundingBox": {
        "lx": 637182.2576560112,
        "ly": 144056.14410144554,
        "lz": 77.62669816520065,
        "ux": 637782.2393459894,
        "uy": 144656.12579142388,
        "uz": 677.6083881435334
    },
    "tightBoundingBox": {
        "lx": 637182.25775,
        "ly": 144056.14425,
        "lz": 77.6267,
        "ux": 637399.95125,
        "uy": 144656.12575,
        "uz": 132.2719
    },
```
parts in the cloud.js, just by looking at it, and create like a cloud2.js or something for the second part, that way you can split off your clouds along the X Y Z planes perhaps, but this... "theory" is untested. i do not really see other ways to do more complex boundaries without re-converting your files.


However i would also add that i am a geologist and not a javascript developer, so perhaps we could wait for someone else to drop by and disprove my statements. :D

**Comment by pewixg3** [comment]
Yes I'll try to enhance the Potree.loadPointCloud function for see all the parts of the pointcloud and I believe that someone more competent can explain us a better solution :D
